portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -13,20 +14,63 @@
 // limitations under the License.
 
 package com.google.sps.servlets;
-
+import java.util.ArrayList;
+import java.util.List;
 import java.io.IOException;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import com.google.gson.Gson;
+import java.time.Instant;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.appengine.api.datastore.DatastoreFailureException;
 
-/** Servlet that returns some example content. TODO: modify this file to handle comments data */
+/** Servlet that handles comments */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
-  @Override
-  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello world!</h1>");
-  }
+    @Override
+    public void doGet(HttpServletRequest req, HttpServletResponse res) throws IOException{
+        List comments = new ArrayList();
+        int limit = Integer.parseInt(req.getParameter("limit"));
+        Query commentsQuery = new Query(ServletUtil.COMMENT_PROPERTY).addSort(ServletUtil.TIMESTAMP_PROPERTY, SortDirection.DESCENDING);
+        PreparedQuery commentResults = ServletUtil.datastore.prepare(commentsQuery);
+        //Integer to keep track of how many comments have been added to the arraylist that will be returned
+        int count = 1;
+        for (Entity commentEntity : commentResults.asIterable()){
+            //Stops iteration after 'limit' iterations
+            if(count++ > limit) {
+               break; 
+            } 
+            //Add's the entity's comment string to the arraylist that will be returned
+            comments.add(commentEntity.getProperty(ServletUtil.COMMENT_PROPERTY));
+        }
+        res.setContentType(ServletUtil.JSON_CONTENT_TYPE);
+
+        res.getWriter().println(ServletUtil.parser.toJson(comments));
+    }
+
+    @Override
+    public void doPost(HttpServletRequest req, HttpServletResponse res) throws IOException{
+        String jsonstring = req.getParameter(ServletUtil.COMMENT_PARAMETER);
```

Comment:
variable name should be camel case, also naming could be commentJson as this json is in relation to that.